### PR TITLE
feat: Celestia DA Layer Client implementation

### PR DESCRIPTION
**Pull request by tzdybal**
This PR introduces celestia DA layer client implementation, based on celestia-node RPC. 

It was tested manually with ephemeral cluster using latest & greatest celestia-node and celestia-app.

Main components:
* `da/celestia` - implementation of DA interface
  * `da/celestia/mock` - mock celestia-node RPC server (for testing)
* `libs/cnrc` - **C**elestia **N**ode **R**PC **C**lient. This is a simple abstraction ti hide complexity related to calling HTTP endpoints, serialization, etc. Right now it's tested via `da/test`. My intention is to extract this a a separate library, because it probably can be reused. Only used methods are implemented.

Resolves: #396

### CODE REVIEW

**Review by jbowen93 — APPROVED** [review]
LGTM. I've tested this against ephemeral cluster myself and tested it with Ethermint. Both work. 

Long term we should consider moving libs/cnrc to a separate repo

Edit: Tomasz is of course ahead of me https://github.com/orgs/celestiaorg/discussions/409

File: libs/cnrc/client.go [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+	err := c.callNamespacedEndpoint(ctx, namespaceID, height, namespacedDataEndpoint, &res)
+	if err != nil {
+		return nil, err
+	}
+
+	return res.Data, nil
+}
+
+// callNamespacedEndpoint fetches result of /namespaced_{type} family of endpoints into result (this should be pointer!)
+func (c *Client) callNamespacedEndpoint(ctx context.Context, namespaceID [8]byte, height uint64, endpoint string, result interface{}) error {
```
Comment (liamsi) [inline libs/cnrc/client.go:99]: Ideally, that 8 would be a const or rather using an existing const.

File: libs/cnrc/types.go [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+// SubmitPFDRequest represents a request to submit a PayForData transaction.
+type SubmitPFDRequest struct {
+	NamespaceID string `json:"namespace_id"`
+	Data        string `json:"data"`
+	GasLimit    uint64 `json:"gas_limit"`
+}
+
+// Types below are copied from celestia-node (or cosmos-sdk dependency of celestia node, to be precise)
+// They are needed because for proper deserialization.
```
Comment (liamsi) [inline libs/cnrc/types.go:13]:
```suggestion
// They are needed for proper deserialization.
```

**Review by liamsi — APPROVED** [review]

File: libs/cnrc/client.go (hunk 1 above)
Comment (tzdybal) [reply libs/cnrc/client.go:99]: Created follow-up issue (#411) as this is used literally everywhere.

**Pull request merged**